test/03-encryption-key.js

```diff
@@ -0,0 +1,28 @@
+const tape = require('tape')
+const crypto = require('crypto')
+const bfe = require('../')
+const { bufferToURIData } = require('../util')
+
+tape('03 encryption key', function (t) {
+  const key = crypto.randomBytes(32)
+
+  const values = [
+    'ssb:diffie-hellman/curve25519/' + bufferToURIData(key),
+    'ssb:diffie-hellman/curve25519/' + bufferToURIData(key) + '?render=thread',
+  ]
+  const valuesLessQueries = values.map((value) => value.replace(/\?.*$/, ''))
```

Comment:
Now that I look at this carefully, what was the intent here? (this is not me asking for a change, just asking a question)

Line 11 adds the `?` and then line 13 removes it

Reply:
valueLessQueries is used to test the decoding down on line 19.

so the query is lost.
What this test specifies is that we support the encoding of URIs which have queries on the end (by throwing it out and just reading the recognised path).

Happy to change that behaviour in future

Reply:
Ahhh, now I see it. My bad, sorry